docker/package/systemd.py

```diff
@@ -0,0 +1,141 @@
+# SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: LicenseRef-MIT-TQ
+from typing import List
+
+# There are more possible fields, but only these are used by tezos services
+class Service:
+    def __init__(
+        self,
+        exec_start: str,
+        state_directory: str,
+        user: str,
+        exec_start_pre: str = None,
+        exec_start_post: str = None,
+        exec_stop_post: str = None,
+        timeout_start_sec: str = None,
+        environment_file: str = None,
+        environment: List[str] = [],
+        remain_after_exit: bool = False,
+        type_: str = None,
+        restart: str = None,
+        keyring_mode: str = None,
+    ):
+        self.environment_file = environment_file
+        self.environment = environment
+        self.exec_start = exec_start
+        self.exec_start_pre = exec_start_pre
+        self.exec_start_post = exec_start_post
+        self.exec_stop_post = exec_stop_post
+        self.timeout_start_sec = timeout_start_sec
+        self.state_directory = state_directory
+        self.user = user
+        self.remain_after_exit = remain_after_exit
+        self.type_ = type_
+        self.restart = restart
+        self.keyring_mode = keyring_mode
+
+
+class Unit:
+    def __init__(
+        self,
+        after: List[str],
+        description: str,
+        requires: List[str] = [],
+        part_of: List[str] = [],
+    ):
+        self.after = after
+        self.requires = requires
+        self.part_of = part_of
+        self.description = description
+
+
+class Install:
+    def __init__(self, wanted_by: List[str] = []):
+        self.wanted_by = wanted_by
+
+
+class ServiceFile:
+    def __init__(self, unit: Unit, service: Service, install: Install):
+        self.unit = unit
+        self.service = service
+        self.install = install
+
+
+class SystemdUnit:
+    def __init__(
+        self,
+        service_file: ServiceFile,
+        startup_script: str = None,
+        startup_script_source: str = None,
+        prestart_script: str = None,
+        prestart_script_source: str = None,
+        suffix: str = None,
+        config_file: str = None,
+        instances: List[str] = [],
+    ):
+        self.suffix = suffix
+        self.service_file = service_file
+        self.startup_script = startup_script
+        self.startup_script_source = startup_script_source
+        self.prestart_script = prestart_script
+        self.prestart_script_source = prestart_script_source
+        self.config_file = config_file
+        self.instances = instances
```

Comment:
This looks kinda ugly with the whole giant `__init__` duplication :disappointed: I see that there's an option to use [`@dataclass`](https://docs.python.org/3/library/dataclasses.html), maybe we would want to use that style to automatically assign fields?

Reply:
:pogchamp:

Reply:
Sadly, Ubuntu 18.04 (whose docker image is used for package building) has python 3.6.9 and dataclasses aren't supported there. But I think it's time to update and use 20.04 instead